### PR TITLE
#3173 - Refresh source immediately

### PR DESCRIPTION
**Pull request by cnevinc**

### CODE REVIEW

File: app/src/main/java/org/mozilla/rocket/content/ContentRepository.kt [hunk 1]
```diff
@@ -28,9 +28,13 @@ class ContentRepository {
 
         @JvmStatic
         fun reset() {
+            INSTANCE?.setOnDataChangedListener(null)
             INSTANCE = null
         }
 
+        @JvmStatic
+        fun empty() = INSTANCE == null
```
Comment (mTwTm) [inline app/src/main/java/org/mozilla/rocket/content/ContentRepository.kt]: isEmpty?

File: app/src/main/java/org/mozilla/rocket/widget/NewsSourcePreference.java [hunk 2]
```diff
@@ -72,6 +73,7 @@ protected void onDialogClosed(boolean positiveResult) {
             Settings.getInstance(getContext()).setPriority(PREF_INT_NEWS_PRIORITY, Settings.PRIORITY_USER);
             setSummary(getValue());
             NewsSourceManager.getInstance().setNewsSource(getValue());
+            ContentRepository.reset();
```
Comment (mTwTm) [inline app/src/main/java/org/mozilla/rocket/widget/NewsSourcePreference.java]: Duplicate call to reset? We already do reset when setNewsSource()

File: app/src/main/java/org/mozilla/focus/home/HomeFragment.java [hunk 3]
```diff
@@ -553,6 +554,18 @@ public void onReceive(Context context, Intent intent) {
         setNightModeEnabled(Settings.getInstance(getActivity()).isNightModeEnable());
         initFeatureSurveyViewIfNecessary(getView());
         playContentPortalAnimation();
+
+        checkNewsRepositoryRested();
+    }
+
+    private void checkNewsRepositoryRested() {
+        if (ContentRepository.empty() && contentViewModel != null && contentPanel != null) {
```
Comment (mTwTm) [inline app/src/main/java/org/mozilla/focus/home/HomeFragment.java]: Maybe some comment here will help.

File: app/src/main/java/org/mozilla/focus/telemetry/TelemetryWrapper.kt [hunk 4]
```diff
@@ -1269,6 +1273,39 @@ object TelemetryWrapper {
...
+            category = Category.ACTION,
+            method = Method.CLICK,
+            `object` = Object.PANEL,
+            value = Value.NEWS,
+            extras = [TelemetryExtra(name = Extra.POSITION, value = "1,2,3..."),
+                    TelemetryExtra(name = Extra.FEED, value = "Newspoint,DainikBhaskar.com"),
+                    TelemetryExtra(name = Extra.SOURCE, value = "India TV,Business World,HW News English...etc"),
+                    TelemetryExtra(name = Extra.CATEGORY, value = "Uttar Pradesh,National,Tech Knowledge....etc"),
+                    TelemetryExtra(name = Extra.SUB_CATEGORY, value = "top-news,entertainment,Lucknow...etc")])
+    fun clickOnNesItem(pos: String, feed: String, source: String, category: String, subCategory: String) {
```
Comment (mTwTm) [inline app/src/main/java/org/mozilla/focus/telemetry/TelemetryWrapper.kt]: typo function name

File: components/partners/partnerrepository/src/main/java/org/mozilla/lite/partner/Repository.java [hunk 5]
```diff
@@ -54,8 +53,12 @@ public void reloadData() {
         nextSubscription();
     }
 
-    public void setOnDataChangedListener(OnDataChangedListener listener) {
+    public void setOnDataChangedListener(@Nullable OnDataChangedListener listener) {
         this.onDataChangedListener = listener;
+        if (listener == null) {
```
Comment (mTwTm) [inline components/partners/partnerrepository/src/main/java/org/mozilla/lite/partner/Repository.java]: Setting listener to null does not always imply a cacheIsDirty. Move
```
itemPojoList = new ArrayList<>();
cacheIsDirty = true;
```
to another function call such as reset();

File: app/src/main/java/org/mozilla/focus/telemetry/TelemetryWrapper.kt (hunk 4 above)
Comment (cnevinc) [reply app/src/main/java/org/mozilla/focus/telemetry/TelemetryWrapper.kt]: will fix in another PR

File: components/partners/partnerrepository/src/main/java/org/mozilla/lite/partner/Repository.java (hunk 5 above)
Comment (cnevinc) [reply components/partners/partnerrepository/src/main/java/org/mozilla/lite/partner/Repository.java]: will fix in another PR

File: components/partners/partnerrepository/src/main/java/org/mozilla/lite/partner/Repository.java [hunk 6]
```diff
@@ -23,7 +23,7 @@
     private Context context;
     private String userAgent;
     private int socketTag;
-    private OnDataChangedListener<T> onDataChangedListener;
+    @Nullable  private OnDataChangedListener<T> onDataChangedListener;
```
Comment (mTwTm) [inline components/partners/partnerrepository/src/main/java/org/mozilla/lite/partner/Repository.java]: extra space?

**Pull request merged**